kernel/src/arch/x86_64/vspace/mod.rs

```diff
@@ -200,10 +201,10 @@ impl Drop for VSpace {
 }
 
 impl VSpace {
-    pub(crate) fn new(da: DA) -> Result<Self, KError> {
+    pub(crate) fn new(allocator: Box<dyn Allocator + Sync + Send>) -> Result<Self, KError> {
```

Comment:
I used Box<dyn Allocator...> to allow for polymorphism (since we now have the DA allocator, and the new ShmemAlloc allocator). Because Allocator trait doesn't have a fixed size, I used Box<dyn...>. But if there's a better/easier way to deal with this in Rust, I'm happy to change it!

Reply:
Ah no that makes sense. I was surprised it works because there used to be a bug which made it such that the allocator has to be a zero-sized type. I looked it up again and it seems they fixed it.